week-4/Homework/mandatory/4-groceries.js

```diff
@@ -39,7 +55,13 @@ Exercise 2:
 */
 // Gather weekend item names into this array
 let weekendGroceriesToBuy = [];
-
+for (day in weeklyMealPlan) {
+  if (day == "saturday" || day == "sunday" && weeklyMealPlan[day].length > 0 ) {   
```

Comment:
This can be both ambiguous and err-prone using `or` and `and` operators on the same level without making the order of precedence explicit. This is something we have yet to teach you all, it is called precedence that determines the evaluation order of expressions.
See more: https://www.dummies.com/web-design-development/javascript-operator-precedence/ 

Your expression is equivalent to this form:
`day == "saturday" || (day == "sunday" && weeklyMealPlan[day].length > 0)`
It means if not `saturday` then condition will be only `true` if day is `sunday and meal plan is empty for that day.
Which is luckily true 😃 in this case. But what if the array weren't empty?

Reply:
Does this make it better ?
if ((day == "saturday" || day == "sunday") && weeklyMealPlan[day].length > 0 )

Reply:
Yes, it is definitely better. Less the chance to fall into a logical pitfall. Parenthesis make the evaluation order of the logical expression unequivocal.

Reply:
BODMAS you mean?

Reply:
Never heard this acronym :) Thanks for that!